### PR TITLE
K8SPXC-1030: Don't delete cert-manager certs by default

### PR DESCRIPTION
**Pull request by egegunes**
[K8SPXC-1030](https://jira.percona.com/browse/K8SPXC-1030) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

We shouldn't set owner references to cert-manager objects if we don't want to delete secrets too. This way, after the PXC cluster is deleted issuers, certificates and their secrets will remain intact in the cluster.

If users want to cleanup objects created for SSL, we introduce a new finalizer: `delete-ssl`. If this finalizer is set, the operator will delete secrets, certificates and issuers. Unfortunately, cert-manager doesn't set owner reference to the secret it created and this behaviour can only configured by command line flag in the controller. Since we can't control how users deploy cert-manager to their clusters, we shouldn't rely on this feature and cleanup certificates and secrets altogether. Hopefully, https://github.com/cert-manager/cert-manager/pull/5158 will merged and we can configure this behaviour on certificate level.

### CODE REVIEW

**Review by tplavcic — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| big-data | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/69a57ced/big-data.log) |
| haproxy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/69a57ced/haproxy.log) |
| storage | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/69a57ced/storage.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/69a57ced/scaling.log) |
| init-deploy | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/69a57ced/init-deploy.log) |
| self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/69a57ced/self-healing.log) |
| scaling-proxysql | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/69a57ced/scaling-proxysql.log) |
| upgrade-haproxy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/69a57ced/upgrade-haproxy.log) |
| recreate | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/69a57ced/recreate.log) |
| scheduled-backup | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/69a57ced/scheduled-backup.log) |
| cross-site | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/69a57ced/cross-site.log) |
| self-healing-chaos | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/69a57ced/self-healing-chaos.log) |
| security-context | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/69a57ced/security-context.log) |
| restore-to-encrypted-cluster | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/69a57ced/restore-to-encrypted-cluster.log) |
| self-healing-advanced | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/69a57ced/self-healing-advanced.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/69a57ced/demand-backup.log) |
| self-healing-advanced-chaos | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/69a57ced/self-healing-advanced-chaos.log) |
| pitr | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/69a57ced/pitr.log) |
| operator-self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/69a57ced/operator-self-healing.log) |
| demand-backup-encrypted-with-tls | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/69a57ced/demand-backup-encrypted-with-tls.log) |
| operator-self-healing-chaos | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/69a57ced/operator-self-healing-chaos.log) |
| upgrade-proxysql | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/69a57ced/upgrade-proxysql.log) |
| smart-update | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/69a57ced/smart-update.log) |
| upgrade-consistency | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/69a57ced/upgrade-consistency.log) |
| limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/69a57ced/limits.log) |
| monitoring-2-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/69a57ced/monitoring-2-0.log) |
| affinity | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/69a57ced/affinity.log) |
| one-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/69a57ced/one-pod.log) |
| auto-tuning | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/69a57ced/auto-tuning.log) |
| proxysql-sidecar-res-limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/69a57ced/proxysql-sidecar-res-limits.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/69a57ced/users.log) |
| tls-issue-self | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/69a57ced/tls-issue-self.log) |
| We run 32 out of 36|

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/1171/commits/69a57cedcde4aefd1a45423259a439ca3953fa87
image: `perconalab/percona-xtradb-cluster-operator:PR-1171-69a57ced`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| haproxy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/69a57ced/haproxy.log) |
| init-deploy | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/69a57ced/init-deploy.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/69a57ced/scaling.log) |
| scheduled-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/69a57ced/scheduled-backup.log) |
| storage | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/69a57ced/storage.log) |
| scaling-proxysql | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/69a57ced/scaling-proxysql.log) |
| self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/69a57ced/self-healing.log) |
| upgrade-haproxy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/69a57ced/upgrade-haproxy.log) |
| recreate | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/69a57ced/recreate.log) |
| big-data | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/69a57ced/big-data.log) |
| cross-site | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/69a57ced/cross-site.log) |
| security-context | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/69a57ced/security-context.log) |
| self-healing-chaos | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/69a57ced/self-healing-chaos.log) |
| restore-to-encrypted-cluster | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/69a57ced/restore-to-encrypted-cluster.log) |
| self-healing-advanced | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/69a57ced/self-healing-advanced.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/69a57ced/demand-backup.log) |
| self-healing-advanced-chaos | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/69a57ced/self-healing-advanced-chaos.log) |
| pitr | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/69a57ced/pitr.log) |
| operator-self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/69a57ced/operator-self-healing.log) |
| demand-backup-encrypted-with-tls | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/69a57ced/demand-backup-encrypted-with-tls.log) |
| operator-self-healing-chaos | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/69a57ced/operator-self-healing-chaos.log) |
| upgrade-proxysql | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/69a57ced/upgrade-proxysql.log) |
| smart-update | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/69a57ced/smart-update.log) |
| upgrade-consistency | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/69a57ced/upgrade-consistency.log) |
| limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/69a57ced/limits.log) |
| monitoring-2-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/69a57ced/monitoring-2-0.log) |
| affinity | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/69a57ced/affinity.log) |
| one-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/69a57ced/one-pod.log) |
| auto-tuning | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/69a57ced/auto-tuning.log) |
| proxysql-sidecar-res-limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/69a57ced/proxysql-sidecar-res-limits.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/69a57ced/users.log) |
| tls-issue-self | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/69a57ced/tls-issue-self.log) |
| tls-issue-cert-manager | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/69a57ced/tls-issue-cert-manager.log) |
| tls-issue-cert-manager-ref | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/69a57ced/tls-issue-cert-manager-ref.log) |
| validation-hook | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/69a57ced/validation-hook.log) |
| proxy-protocol | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/69a57ced/proxy-protocol.log) |
| We run 36 out of 36|

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/1171/commits/69a57cedcde4aefd1a45423259a439ca3953fa87
image: `perconalab/percona-xtradb-cluster-operator:PR-1171-69a57ced`

**Review by hors — CHANGES_REQUESTED** [review]
@egegunes please fix the tests and add the test case for new finalizer

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/scaling.log) |
| storage | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/storage.log) |
| big-data | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/big-data.log) |
| upgrade-haproxy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/upgrade-haproxy.log) |
| cross-site | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/cross-site.log) |
| recreate | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/recreate.log) |
| scheduled-backup | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/scheduled-backup.log) |
| haproxy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/haproxy.log) |
| scaling-proxysql | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/scaling-proxysql.log) |
| self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/self-healing.log) |
| self-healing-chaos | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/self-healing-chaos.log) |
| restore-to-encrypted-cluster | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/restore-to-encrypted-cluster.log) |
| upgrade-proxysql | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/upgrade-proxysql.log) |
| security-context | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/security-context.log) |
| self-healing-advanced | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/self-healing-advanced.log) |
| self-healing-advanced-chaos | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/self-healing-advanced-chaos.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/init-deploy.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/demand-backup.log) |
| smart-update | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/smart-update.log) |
| operator-self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/operator-self-healing.log) |
| operator-self-healing-chaos | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/operator-self-healing-chaos.log) |
| limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/limits.log) |
| pitr | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/pitr.log) |
| monitoring-2-0 | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/monitoring-2-0.log) |
| upgrade-consistency | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/upgrade-consistency.log) |
| affinity | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/affinity.log) |
| demand-backup-encrypted-with-tls | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/demand-backup-encrypted-with-tls.log) |
| one-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/one-pod.log) |
| auto-tuning | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/auto-tuning.log) |
| proxysql-sidecar-res-limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/proxysql-sidecar-res-limits.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/users.log) |
| tls-issue-self | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/tls-issue-self.log) |
| tls-issue-cert-manager | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/tls-issue-cert-manager.log) |
| tls-issue-cert-manager-ref | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/tls-issue-cert-manager-ref.log) |
| validation-hook | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/validation-hook.log) |
| proxy-protocol | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/proxy-protocol.log) |
| We run 36 out of 36|

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/1171/commits/658e73dfc00be8cb1428da3a883ca27c0d2fc53f
image: `perconalab/percona-xtradb-cluster-operator:PR-1171-658e73df`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scheduled-backup | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/scheduled-backup.log) |
| haproxy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/haproxy.log) |
| storage | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/storage.log) |
| upgrade-haproxy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/upgrade-haproxy.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/scaling.log) |
| cross-site | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/cross-site.log) |
| recreate | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/recreate.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/init-deploy.log) |
| self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/self-healing.log) |
| big-data | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/big-data.log) |
| scaling-proxysql | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/scaling-proxysql.log) |
| restore-to-encrypted-cluster | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/restore-to-encrypted-cluster.log) |
| self-healing-chaos | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/self-healing-chaos.log) |
| limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/limits.log) |
| security-context | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/security-context.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/demand-backup.log) |
| self-healing-advanced | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/self-healing-advanced.log) |
| monitoring-2-0 | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/monitoring-2-0.log) |
| pitr | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/pitr.log) |
| self-healing-advanced-chaos | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/self-healing-advanced-chaos.log) |
| operator-self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/operator-self-healing.log) |
| operator-self-healing-chaos | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/operator-self-healing-chaos.log) |
| upgrade-proxysql | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/upgrade-proxysql.log) |
| smart-update | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/smart-update.log) |
| upgrade-consistency | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/upgrade-consistency.log) |
| affinity | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/affinity.log) |
| one-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/one-pod.log) |
| auto-tuning | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/auto-tuning.log) |
| proxysql-sidecar-res-limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/proxysql-sidecar-res-limits.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/users.log) |
| tls-issue-self | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/tls-issue-self.log) |
| tls-issue-cert-manager | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/tls-issue-cert-manager.log) |
| tls-issue-cert-manager-ref | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/tls-issue-cert-manager-ref.log) |
| validation-hook | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/validation-hook.log) |
| proxy-protocol | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/proxy-protocol.log) |
| demand-backup-encrypted-with-tls | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/demand-backup-encrypted-with-tls.log) |
| We run 36 out of 36|

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/1171/commits/658e73dfc00be8cb1428da3a883ca27c0d2fc53f
image: `perconalab/percona-xtradb-cluster-operator:PR-1171-658e73df`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| recreate | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/recreate.log) |
| big-data | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/big-data.log) |
| restore-to-encrypted-cluster | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/restore-to-encrypted-cluster.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/scaling.log) |
| scheduled-backup | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/scheduled-backup.log) |
| cross-site | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/cross-site.log) |
| haproxy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/haproxy.log) |
| upgrade-haproxy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/upgrade-haproxy.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/demand-backup.log) |
| scaling-proxysql | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/scaling-proxysql.log) |
| storage | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/storage.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/init-deploy.log) |
| pitr | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/pitr.log) |
| security-context | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/security-context.log) |
| limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/limits.log) |
| self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/self-healing.log) |
| demand-backup-encrypted-with-tls | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/demand-backup-encrypted-with-tls.log) |
| monitoring-2-0 | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/monitoring-2-0.log) |
| self-healing-chaos | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/self-healing-chaos.log) |
| self-healing-advanced | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/self-healing-advanced.log) |
| self-healing-advanced-chaos | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/self-healing-advanced-chaos.log) |
| operator-self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/operator-self-healing.log) |
| operator-self-healing-chaos | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/operator-self-healing-chaos.log) |
| upgrade-proxysql | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/upgrade-proxysql.log) |
| smart-update | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/smart-update.log) |
| upgrade-consistency | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/upgrade-consistency.log) |
| affinity | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/affinity.log) |
| one-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/one-pod.log) |
| auto-tuning | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/auto-tuning.log) |
| proxysql-sidecar-res-limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/proxysql-sidecar-res-limits.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/users.log) |
| tls-issue-self | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/tls-issue-self.log) |
| tls-issue-cert-manager | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/tls-issue-cert-manager.log) |
| tls-issue-cert-manager-ref | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/tls-issue-cert-manager-ref.log) |
| validation-hook | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/validation-hook.log) |
| proxy-protocol | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1171/658e73df/proxy-protocol.log) |
| We run 36 out of 36|

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/1171/commits/658e73dfc00be8cb1428da3a883ca27c0d2fc53f
image: `perconalab/percona-xtradb-cluster-operator:PR-1171-658e73df`

File: pkg/controller/pxc/controller.go [hunk 1]
```diff
@@ -228,6 +229,8 @@ func (r *ReconcilePerconaXtraDBCluster) Reconcile(_ context.Context, request rec
 		for _, fnlz := range o.GetFinalizers() {
 			var sfs api.StatefulApp
 			switch fnlz {
+			case "delete-ssl":
```
Comment (nmarukovich) [inline pkg/controller/pxc/controller.go:232]: should we add it to cr.yml as an example?

**Comment by nmarukovich** [comment]
with 2.28.0 pmm-client version monitoring test passed.

**Review by hors — APPROVED** [review]

File: pkg/controller/pxc/controller.go [hunk 2]
```diff
@@ -1172,6 +1175,67 @@ func (r *ReconcilePerconaXtraDBCluster) deleteSecrets(cr *api.PerconaXtraDBClust
...
+	certs := []string{
+		cr.Name + "-ssl",
+		cr.Name + "-ssl-internal",
+		cr.Name + "-ca-cert",
+	}
+	for _, certName := range certs {
+		secret := &corev1.Secret{}
+		err := r.client.Get(context.TODO(), types.NamespacedName{
+			Namespace: cr.Namespace,
+			Name:      certName,
```
Comment (pooknull) [inline pkg/controller/pxc/controller.go:1208]: Secrets can have a different name if `sslSecretName` or `sslInternalSecretName` in provided cr

Comment (hors) [reply pkg/controller/pxc/controller.go:1208]: @egegunes please check it and fix

**Pull request merged**